include/merlin/group_lock.hpp

```diff
@@ -0,0 +1,213 @@
+/*
```

Comment:
Hi @zhangyafeikimi , could you help review on this group lock, thank you so much!

Reply:
have given some comments

Reply:
Accept.